make.config.in

```diff
@@ -189,14 +197,16 @@ ifneq ("$(TARGET)", "runtest")
 ####################################################################
 # Make libbout++.a if it doesn't exist with the checklib target
 ####################################################################
-%.a:
+libbout++.a:
 	@echo "Rebuilding out-of-date bout++ library"
 	@$(MAKE) --no-print-directory -C $(BOUT_TOP)
  
 ####################################################################
 # Make the target (e.g. gas_compress)
 ####################################################################
-SUB_LIBS=$(DIRS:%=%.a)
+DIRS_=$(DIRS:%/=%)
+DIRS__=$(shell for f in $(DIRS_) ; do echo $${f\#\#*/};done)
```

Comment:
Another comment for these two lines as well please. Also, why the `#`s need escaping here and not above.